/*
* Copyright (c) 2015 Qualcomm Atheros, Inc.
* All Rights Reserved.
* Qualcomm Atheros Confidential and Proprietary.
* NOT A CONTRIBUTION
*/

diff -Nuar a/kernel/arch/x86/configs/android-x86_defconfig b/kernel/arch/x86/configs/android-x86_defconfig
--- a/kernel/arch/x86/configs/android-x86_defconfig	2015-06-16 16:01:29.844605255 +0800
+++ b/kernel/arch/x86/configs/android-x86_defconfig	2015-06-16 17:18:03.396595551 +0800
@@ -1052,7 +1052,7 @@
 #
 # Bluetooth device drivers
 #
-CONFIG_BT_HCIBTUSB=m
+#CONFIG_BT_HCIBTUSB=m
 CONFIG_BT_HCIBTSDIO=m
 CONFIG_BT_HCIUART=m
 CONFIG_BT_HCIUART_H4=y
@@ -1128,7 +1128,7 @@
 # Generic Driver Options
 #
 CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
-# CONFIG_DEVTMPFS is not set
+CONFIG_DEVTMPFS=y
 CONFIG_STANDALONE=y
 CONFIG_PREVENT_FIRMWARE_BUILD=y
 CONFIG_FW_LOADER=y
@@ -2038,6 +2038,7 @@
 #
 # USB Network Adapters
 #
+CONFIG_SS1BTUSB=m
 CONFIG_USB_CATC=m
 CONFIG_USB_KAWETH=m
 CONFIG_USB_PEGASUS=m
diff -Nuar a/kernel/drivers/usb/misc/Kconfig b/kernel/drivers/usb/misc/Kconfig
--- a/kernel/drivers/usb/misc/Kconfig	2015-06-16 16:01:23.716605268 +0800
+++ b/kernel/drivers/usb/misc/Kconfig	2015-06-16 17:18:03.404595551 +0800
@@ -3,6 +3,17 @@
 #
 comment "USB Miscellaneous drivers"
 
+config SS1BTUSB
+	tristate "SS1 BT USB driver"
+	depends on USB
+	help
+	  Bluetooth SS1 BT USB driver.
+	  This driver is required if you want to use Bluetooth devices with
+	  USB interface.
+
+	  Say Y here to compile support for Bluetooth USB devices into the
+	  kernel or say M to compile it as module (SS1BTUSB).
+
 config USB_EMI62
 	tristate "EMI 6|2m USB Audio interface support"
 	---help---
diff -Nuar a/kernel/drivers/usb/misc/Makefile b/kernel/drivers/usb/misc/Makefile
--- a/kernel/drivers/usb/misc/Makefile	2015-06-16 16:01:23.712605268 +0800
+++ b/kernel/drivers/usb/misc/Makefile	2015-06-16 17:18:03.404595551 +0800
@@ -5,6 +5,7 @@
 
 ccflags-$(CONFIG_USB_DEBUG) := -DDEBUG
 
+obj-$(CONFIG_SS1BTUSB)			+= SS1BTUSB.o
 obj-$(CONFIG_USB_ADUTUX)		+= adutux.o
 obj-$(CONFIG_USB_APPLEDISPLAY)		+= appledisplay.o
 obj-$(CONFIG_USB_CYPRESS_CY7C63)	+= cypress_cy7c63.o
diff -Naur a/kernel/drivers/hid/hid-input.c b/kernel/drivers/hid/hid-input.c
--- a/kernel/drivers/hid/hid-input.c	2015-10-21 19:30:02.201725337 +0800
+++ b/kernel/drivers/hid/hid-input.c	2015-10-21 19:33:18.101724124 +0800
@@ -1378,9 +1378,14 @@
 	}
 
 	if (hidinput) {
+/* 2015/10/21
+ * temp fix for apple magic mouse unable to move issue 
+ */
+#if 0
 		if (drv->input_configured &&
 		    drv->input_configured(hid, hidinput))
 			goto out_cleanup;
+#endif
 		if (input_register_device(hidinput->input))
 			goto out_cleanup;
 	}
